File: 프로그래머스/unrated/181859. 배열 만들기 6/배열 만들기 6.java
class Solution {
    public int[] solution(int[] arr) {
        int len = arr.length;
        int[] src = new int[len];
        int count = 0;
        for (int i = 0; i < arr.length; i++)
        {
            if (0 < count && src[count - 1] == arr[i])
                count--;
            else
                src[count++] = arr[i];    
        }
        if (count == 0)
            src[count++] = -1;
        int[] answer = new int[count];
        System.arraycopy(src, 0, answer, 0, count);
        return answer;
    }
}

File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.js
function solution(arr) {
    const len = arr.length;
    var stk = [];
    for (var i = 0; i < len; i++) {
        if (stk.length === 0)
            stk.push(arr[i]);
        else if (stk[stk.length - 1] === arr[i])
            stk.pop();
        else
            stk.push(arr[i]);
    }
    return stk.length ? stk : [-1];
}